On macos 12 x86_64, `folly` fails to build:

```
In file included from /tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/logging/example/lib.cpp:17:
In file included from /tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/logging/example/lib.h:21:
In file included from /tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/logging/xlog.h:26:
In file included from /tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/logging/LoggerDB.h:29:
In file included from /tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/detail/StaticSingletonManager.h:27:
In file included from /tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/lang/Thunk.h:22:
/tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/lang/New.h:66:12: error: no matching function for call to object of type 'const conditional_t<false, op_del_builtin_fn_, op_del_library_fn_>' (aka 'const folly::detail::op_del_library_fn_')
    return op_del_<Usual>(p, a...);
           ^~~~~~~~~~~~~~
/tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/lang/New.h:119:20: note: in instantiation of function template specialization 'folly::detail::do_op_del_sized_<false, std::align_val_t>' requested here
    return detail::do_op_del_sized_<detail::cpp_aligned_new_>(p, s, a);
                   ^
/tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/lang/New.h:52:1: note: candidate template ignored: substitution failure [with A = <void *const &, const std::align_val_t &>]
FOLLY_CREATE_QUAL_INVOKER(op_del_library_fn_, ::operator delete);
^
/tmp/nix-build-folly-2024.03.11.00.drv-0/source/folly/functional/Invoke.h:515:58: note: expanded from macro 'FOLLY_CREATE_QUAL_INVOKER'
    [[maybe_unused]] FOLLY_ERASE_HACK_GCC constexpr auto operator()(A&&... a) \
```

--- a/pkgs/development/libraries/fbthrift/default.nix
+++ b/pkgs/development/libraries/fbthrift/default.nix
@@ -23,13 +23,13 @@
 
 stdenv.mkDerivation rec {
   pname = "fbthrift";
-  version = "2024.01.22.00";
+  version = "2024.03.11.00";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "fbthrift";
     rev = "v${version}";
-    sha256 = "sha256-vIYXX4NOs2JdhrAJKmIhf4+hQEXHue2Ok7e4cw6yups=";
+    sha256 = "sha256-iCiiKNDlfKm1Y4SGzcSP6o/OdiRRrj9UEawW6qpBpSY=";
   };
 
   nativeBuildInputs = [
@@ -38,7 +38,9 @@ stdenv.mkDerivation rec {
     flex
   ];
 
-  cmakeFlags = lib.optionals stdenv.isDarwin [
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=${if stdenv.isDarwin then "OFF" else "ON"}"
+  ] ++ lib.optionals stdenv.isDarwin [
     "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.14" # For aligned allocation
   ];
 

--- a/pkgs/development/libraries/fizz/default.nix
+++ b/pkgs/development/libraries/fizz/default.nix
@@ -19,13 +19,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "fizz";
-  version = "2024.01.22.00";
+  version = "2024.03.11.00";
 
   src = fetchFromGitHub {
     owner = "facebookincubator";
     repo = "fizz";
     rev = "refs/tags/v${finalAttrs.version}";
-    hash = "sha256-17EELvRrWhUprxvm1Ur0FYNimvY1qgK0YH8ehxtLpxM=";
+    hash = "sha256-IHWotiVUjGOvebXy4rwsh8U8UMxTrF1VaqXzZMjojiM=";
   };
 
   nativeBuildInputs = [ cmake ];

--- a/pkgs/development/libraries/folly/default.nix
+++ b/pkgs/development/libraries/folly/default.nix
@@ -1,5 +1,6 @@
 { lib
 , stdenv
+, overrideSDK
 , fetchFromGitHub
 , boost
 , cmake
@@ -26,13 +27,13 @@
 
 stdenv.mkDerivation rec {
   pname = "folly";
-  version = "2024.01.22.00";
+  version = "2024.03.11.00";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "folly";
     rev = "v${version}";
-    sha256 = "sha256-+z1wuEOgr7CMHFnOn5gLm9mtVH7mVURLstOoDqzxKbk=";
+    sha256 = "sha256-INvWTw27fmVbKQIT9ebdRGMCOIzpc/NepRN2EnKLJx0=";
   };
 
   nativeBuildInputs = [
@@ -72,6 +73,8 @@ stdenv.mkDerivation rec {
     # see https://github.com/NixOS/nixpkgs/issues/144170
     "-DCMAKE_INSTALL_INCLUDEDIR=include"
     "-DCMAKE_INSTALL_LIBDIR=lib"
+  ] ++ lib.optional (stdenv.isDarwin && stdenv.isx86_64) [
+    "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.13"
   ];
 
   # split outputs to reduce downstream closure sizes

